staging/src/k8s.io/kubectl/pkg/cmd/drain/drain.go

```diff
@@ -285,6 +287,28 @@ func (o *DrainCmdOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args [
 	})
 }
 
+func (o *DrainCmdOptions) drainNode(
+	nodeInfo *resource.Info,
+	drainResultMutex *sync.Mutex,
+	drainWG *sync.WaitGroup,
+	drainedNodes *sets.String,
+	drainErrors *[]error,
+	printObj printers.ResourcePrinterFunc,
+) {
```

Comment:
I'd prefer we split the parallelism part (workgroups & mutexes) outside this function, such that this function is only responsible for draining a node. The entire wrapper should be handled either inside a separate function or inside that `if-else` condition below.

Reply:
The difficulty is that `drainNode` is being invoked in a go-routine, so the caller can't block waiting for it to return to update mutexes and WGs. I initially added this `drainNode` function to handle that parallelism book-keeping in a single place.

Would renaming `drainNode` make its book-keeping role more clear? Or do you think I should wrap `drainNode` in another function that will be invoked in the go-routine and will handle the book-keeping?